daemon/delete.go

```diff
@@ -76,22 +76,20 @@ func (daemon *Daemon) rm(container *Container, forceRemove bool) (err error) {
 		}
 	}
 
-	// stop collection of stats for the container regardless
-	// if stats are currently getting collected.
-	daemon.statsCollector.stopCollection(container)
```

Comment:
I moved this after setting the removal state because there is no point to try to stop collecting stats from a container that has already being marked for removal. This never failed though because in that case, `stopCollection` is a noop.